### PR TITLE
Refactor to use guard clause

### PR DESCRIPTION
**Pull request by nikkypx**
Method does not need one exit, this give a clean escape immediately

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -38,12 +38,9 @@ def user_signed_in?
...
   def authenticate_user!
-    if current_user
-      true
-    else
-      session["return_to"] ||= request.url
-      redirect_to login_path unless login_redirect? or oauth_callback?
-    end
+    return true if current_user
+    session["return_to"] ||= request.url
+    redirect_to login_path unless login_redirect? or oauth_callback?
```
Comment (fusion2004) [inline app/controllers/application_controller.rb]: Shouldn't this be doing: `unless login_redirect? || oauth_callback?`

Comment (olivierlacan) [inline app/controllers/application_controller.rb]: Please don't use `unless` in authentication flows, it's a recipe for bad news. :-)

Comment (olivierlacan) [inline app/controllers/application_controller.rb]: And yes I'm aware it was already used. :cry:

Comment (nikkypx) [inline app/controllers/application_controller.rb]: it looks like the unless can just be taken off there, and just redirect_to login_path, but i'm not sure if that breaks anything

Comment (olivierlacan) [inline app/controllers/application_controller.rb]: @nikkypx No, if we do that we could get infinite redirect loops if the `session["return_to"]` was the `login_path`. I'll fix @fusion2004's suggestion separately.

**Comment by olivierlacan** [comment]
@fusion2004 See c044ed718a2bbb83b9f2599b37a635dfa55a186c

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (nikkypx) [inline app/controllers/application_controller.rb]: ah right

**Pull request merged**